BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m

```diff
@@ -99,15 +99,15 @@ - (void)performAuthorizationCodeGrantWithReceivedURL:(NSURL *)URL withCompletion
         return;
     }
 
-
-    NSDictionary *POSTParams = @{
-                                 BOXAuthTokenRequestGrantTypeKey : BOXAuthTokenRequestGrantTypeAuthorizationCode,
-                                 BOXAuthTokenRequestAuthorizationCodeKey : authorizationCode,
-                                 BOXAuthTokenRequestClientIDKey : self.clientID,
-                                 BOXAuthTokenRequestClientSecretKey : self.clientSecret,
-                                 BOXAuthTokenRequestRedirectURIKey : self.redirectURIString,
-                                 };
-
+    NSMutableDictionary *POSTParams = [NSMutableDictionary dictionaryWithDictionary:@{
+                                                                                      BOXAuthTokenRequestGrantTypeKey : BOXAuthTokenRequestGrantTypeAuthorizationCode,
+                                                                                      BOXAuthTokenRequestAuthorizationCodeKey : authorizationCode,
+                                                                                      BOXAuthTokenRequestClientIDKey : self.clientID,
+                                                                                      BOXAuthTokenRequestClientSecretKey : self.clientSecret,
+                                                                                      BOXAuthTokenRequestRedirectURIKey : self.redirectURIString,
+                                                                                      }];
+    [POSTParams addEntriesFromDictionary:self.additionalTokenGrantParams];
```

Comment:
it is nil safe. But decided to introduce the check for existence of contents within the dictionary before adding.